src/Server.ts

```diff
@@ -123,6 +124,13 @@ export default class Server {
     response: http.ServerResponse,
   ): Promise<void> {
     let body: Uint8Array[];
+
+    /* Allow cross origin request, this is needed to build an UI on top of faucet */
```

Comment:
We should think about this if we want to allow the cross-origin request.
Maybe we can deploy the UI at a particular domain and just allow access for that domain.

Reply:
Shouldn't the headers only be sent in response to an `OPTIONS` request? [Preflight requests in CORS on MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS#Preflighted_requests_in_CORS)

Reply:
@schemar I am always confused on this topic. Please check the second para on the link, it says all the methods.  Though I am not sure about this. 

> The server responds with Access-Control-Allow-Methods and says that POST, GET, and OPTIONS are viable methods to query the resource in question. This header is similar to the Allow response header, but used strictly within the context of CORS.

Reply:
You make an `OPTIONS` **request**. In the **response** to *that* request, the server tells the client which methods are allowed (for *future* requests), e.g. `OPTIONS` and `GET`.